### PR TITLE
Tabbed navigation for Appeals

### PR DESCRIPTION
**Pull request by markgreenburg**
To Do:
[x] - Set up routing & scaffolding for appeal, appeal status, appeal details
[x] - Set up tabs: set active link by default, navigate between tabs w/functional history.push()
[x] - Conditionally render content based on active tab

Split off into other PRs:
- Better formatting to more closely match layout in mocks
- CSS: 1. convert reused claims-styles to appeals-styles for better control, 2. add styles for things that don't look right to match the mocks

### CODE REVIEW

File: src/js/claims-status/containers/AppealsV2DetailPage.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const AppealsV2DetailPage = () => {
+  return (
+    <div>
+      Hey I'm the details page!
```
Comment (cvalarida) [inline src/js/claims-status/containers/AppealsV2DetailPage.jsx]: 😂

File: src/js/claims-status/containers/AppealInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    </div>
+  );
+};
+
+AppealInfo.PropTypes = {
+  params: PropTypes.shape({ id: PropTypes.string.isRequired }).isRequired,
+  route: PropTypes.shape({ path: PropTypes.string.isRequired }).isRequired,
+};
+
+export default AppealInfo;
```
Comment (cvalarida) [inline src/js/claims-status/containers/AppealInfo.jsx]: Could you clarify why you export this twice--once as a named export and once as the default?

File: src/js/claims-status/routes.jsx [hunk 3]
```diff
@@ -41,10 +42,19 @@ const routes = [
...
+      key=":id"
+      from=":id"
+      to=":id/status"/>
     <Route
-      component={AppealsV2StatusPage}
+      component={AppealInfo}
       key=":id/status"
       path=":id/status"/>
+    <Route
+      component={AppealInfo}
```
Comment (cvalarida) [inline src/js/claims-status/routes.jsx]: Can we wrap these `Route`s in another `Route` that uses `AppealInfo` as its component and have that just renders its children?

```jsx
<Route
  component={AppealInfo}
  key=":id"
  path=":id">
  <Redirect
    key="/"
    from="/" {/* May need to be an IndexRedirect or something; I don't remember */}
    to="/status"/>
  <Route
    component={AppealsV2StatusPage}
    key="/status"
    path="/status"/>
  <Route
    component={AppealsV2DetailsPage}
    key="/details"
    path="/details"/>
</Route>
```

This way, we won't have to do any route-based rendering from inside `AppealInfo`, y'know?

File: src/js/claims-status/containers/AppealInfo.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import AppealsV2DetailPage from './AppealsV2DetailPage';
+import AppealsV2TabNav from '../components/appeals-v2/AppealsV2TabNav';
+
+const PATHS = { status: 'status', detail: 'detail' };
+
+const setActiveTab = (currentPath, paths = { status: '', detail: '' }, appealId) => {
+  if (currentPath.includes(paths.status)) {
+    return <AppealsV2StatusPage appealId={appealId}/>;
+  } else if (currentPath.includes(paths.detail)) {
+    return <AppealsV2DetailPage appealId={appealId}/>;
```
Comment (cvalarida) [inline src/js/claims-status/containers/AppealInfo.jsx]: 🤔  I'm not super keen on this bit; see my comment in `routes.jsx`. I think we can get around this with a bit more fancy footwork in `react-router`.

**Review by cvalarida — COMMENTED** [review]
It looks pretty good, but can we use `react-router` to handle rendering the status and details page? This works, but looking at it later might get confusing.

File: src/js/claims-status/containers/AppealInfo.jsx (hunk 2 above)
Comment (markgreenburg) [reply src/js/claims-status/containers/AppealInfo.jsx]: Yeah, I got rid of these. Generally I export twice to isolate e.g. connected components in unit tests. Obviously that's not needed here. Not to mention, it broke the tests.

File: src/js/claims-status/containers/AppealsV2DetailPage.jsx (hunk 1 above)
Comment (markgreenburg) [reply src/js/claims-status/containers/AppealsV2DetailPage.jsx]: Finalize content: ✓

File: src/js/claims-status/routes.jsx (hunk 3 above)
Comment (markgreenburg) [reply src/js/claims-status/routes.jsx]: 🤔 Let me look into that. I'm all for reducing complexity in the components.

File: src/js/claims-status/containers/AppealInfo.jsx (hunk 4 above)
Comment (markgreenburg) [reply src/js/claims-status/containers/AppealInfo.jsx]: I decided to keep things consistent-ish with what was done for Claims before but...I think I like the router-based implementation better. Will play around and see what I can do.

**Review by cvalarida — APPROVED** [review]
Beautiful!

**Pull request merged**